app/src/main/java/com/example/music/MusicList.java

```diff
@@ -109,40 +112,43 @@ public void onPermissionRationaleShouldBeShown(PermissionRequest permission, Per
                 selection,
                 null,
                 null);
-
-        ArrayList<String> songs = new ArrayList<String>();
+        int idColumn = cursor.getColumnIndexOrThrow(MediaStore.Audio.Media._ID);
+        ArrayList<Uri> uris = new ArrayList<>();
         while (cursor.moveToNext()) {
+            long id = cursor.getLong(idColumn);
             int time = Integer.parseInt(cursor.getString(5));
             time = time / (1000 * 60);
-            songs.add(cursor.getString(1) + " " + cursor.getString(2) + " " + cursor.getString(4) + " " + time);
+            items.add(cursor.getString(1) + " " + cursor.getString(2) + " " + cursor.getString(4) + " " + time + " ");
+            Uri contentUri = ContentUris.withAppendedId(
+                    MediaStore.Audio.Media.EXTERNAL_CONTENT_URI, id);
+            uris.add(contentUri);
         }
-        return songs;
+        cursor.close();
+        return uris;
     }
 
 
     public void displaySongs() {
+        ArrayList<Uri> uris = new ArrayList<>();
         try {
-//            Environment.getExternalStorageDirectory()
-//            final ArrayList<String> mySongs = findSong();
-            items = findSong();
+            uris = findSong();
             ArrayAdapter<String> myAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, items);
             listView.setAdapter(myAdapter);
         } catch (Exception e) {
             e.printStackTrace();
         }
 
-      //Ye video se dekha hai hai ispr errors aray hain
-
-          /*  listView.setOnClickListener(new AdapterView.OnItemClickListener() {
-                @Override
-                public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
-                        String songName = (String)listView.getItemAtPosition(position);
-                        startActivity(new Intent(getApplicationContext(),Activityplayer.class).putExtra("songs", mySong).putExtra("songname",songName).putExtra("pos", position));
-
-                }*/
-    }//);
-
+        //Fixed by subhan
```

Comment:
Please remove this, as the commits already highlight which person worked on what feature.